electron/electron.js

```diff
@@ -74,3 +74,24 @@ ipcMain.handle('get-file', async () =>
     properties: ['openFile'],
   }),
 );
+
+ipcMain.handle('get-output-path', async () => {
+  const options = {
+    title: 'Select Output Folder',
+    defaultPath: app.getPath('downloads'),
+    properties: ['openDirectory', 'createDirectory'],
+  };
+  return dialog.showOpenDialog(null, options, (savePath) => savePath);
+});
+
+ipcMain.handle('save-output', async (event, savePath, extractedData) => {
+  try {
+    extractedData.forEach((bundle, i) => {
+      const outputFile = path.join(savePath, `mcode-extraction-patient-${i + 1}.json`);
+      fs.writeFileSync(outputFile, JSON.stringify(bundle), 'utf8');
+    });
+    return true;
+  } catch (error) {
+    return false;
+  }
+});
```

Comment:
I think the flow can probably be improved here such that in the error cases, we send some time of error message back to the client so the user is aware that an issue happened. 

Perhaps this return value can be changed such that `error.message` makes its way back to the react app? And then instead of just rendering "output saved" or not, we could render the error message in the error case